src/main/java/org/spongepowered/common/mixin/core/world/biome/MixinBiomeMesa.java

```diff
@@ -95,7 +97,9 @@ public void buildPopulators(World world, SpongeBiomeGenerationSettings gensettin
      */
     @Inject(method = "genTerrainBlocks(Lnet/minecraft/world/World;Ljava/util/Random;Lnet/minecraft/world/chunk/ChunkPrimer;IID)V", at = @At("HEAD") , cancellable = true)
     public void genTerrainBlocks(World world, Random rand, ChunkPrimer chunk, int x, int z, double stoneNoise, CallbackInfo ci) {
-        ci.cancel();
+        if (this.vanillaCompat) {
```

Comment:
@Deamon5550 I've spent a couple of hours revisiting this, Any tips on how to get the generator from inside this injection?

Reply:
@what you could do is take advantage of the `PhaseTracker` and have a boolean flag for whether this biome is usable or not (since world generation is still tracked through a phase state).